### PR TITLE
fix(Charts): removed theme variant

### LINKED ISSUE #232: Charts - removed theme variant for v5
Follow up to breaking change PR https://github.com/patternfly/patternfly-react/pull/8590

We've removed the deprecated `themeVariant` property from `getCustomTheme` -- currently documented in the REAME. This functionality was previously a noop and replaced by PatternFly core's dark theme support.

We also removed the `themeVariant` property from the components below.

- Chart
- ChartArea
- ChartAxis
- ChartBar
- ChartBullet
- ChartBulletComparativeErrorMeasure
- ChartBulletComparativeMeasure
- ChartBulletComparativeWarningMeasure
- ChartBulletGroupTitle
- ChartBulletPrimaryDotMeasure
- ChartBulletPrimarySegmentedMeasure
- ChartBulletQualitativeRange
- ChartBulletTitle
- ChartContainer
- ChartCursorContainer
- ChartCursorTooltip
- ChartDonut
- ChartDonutThreshold
- ChartDonutUtilization
- ChartGroup/ChartGroup
- ChartLegend/ChartLegend
- ChartLegendTooltip
- ChartLegendTooltipContent
- ChartLine/ChartLine
- ChartPie/ChartPie
- ChartScatter/ChartScatter
- ChartStack/ChartStack
- ChartTheme/ChartTheme
- ChartThreshold/ChartThreshold
- ChartTooltip/ChartTooltip
- ChartVoronoiContainer

The components are our documented public APIs, while helper utils remain private unless otherwise stated in the component docs. However, it may be worth noting that the following unused, dark theme objects have been removed.

- DarkBlueColorTheme
- DarkCyanColorTheme
- DarkGoldColorTheme
- DarkGrayColorTheme
- DarkGreenColorTheme
- DarkMultiColorOrderedTheme
- DarkMultiColorUnorderedTheme
- DarkOrangeColorTheme
- DarkPurpleColorTheme

In addition, the following light theme objects have been renamed and marked `@private` for clarity. Developers shouldn't reference objects directly, but use `getTheme(ChartThemeColor.default)` instead. 

- LightBlueColorTheme as BlueColorTheme
- LightCyanColorTheme as CyanColorTheme
- LightGoldColorTheme as GoldColorTheme
- LightGrayColorTheme as GrayColorTheme
- LightGreenColorTheme as GreenColorTheme
- LightMultiColorOrderedTheme as MultiColorOrderedTheme
- LightMultiColorUnorderedTheme as MultiColorUnorderedTheme
- LightOrangeColorTheme as OrangeColorTheme
- LightPurpleColorTheme as PurpleColorTheme

Required actions:
1. Build codemod
2. Build test
3. Updated README via PR https://github.com/patternfly/pf-codemods/pull/227

### PR DESCRIPTION
**Pull request by thatblindgeye**
Closes #232

### CODE REVIEW

**Review by dlabrecq — APPROVED** [review]

File: packages/eslint-plugin-pf-codemods/lib/rules/v5/charts-remove-themeVariant.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            (attribute) => attribute.name.name === "themeVariant"
+          );
+
+          if (themeVariantProp) {
+            context.report({
+              node,
+              message:
+                "The themeVariant prop has been removed for all react-charts components.",
+              fix(fixer) {
+                return fixer.replaceTextRange(themeVariantProp.range, "");
```
Comment (gitdallas) [inline packages/eslint-plugin-pf-codemods/lib/rules/v5/charts-remove-themeVariant.js:28]:
```suggestion
                const prevToken = context.getSourceCode().getTokenBefore(themeVariantProp);
                return fixer.replaceTextRange([prevToken.range[1], themeVariantProp.range[1]], "");
```

File: packages/eslint-plugin-pf-codemods/test/rules/v5/charts-remove-themeVariant.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    },
+    // No @patternfly/react-charts import
+    {
+      code: `<Chart themeVariant />`,
+    },
+  ],
+  invalid: [
+    {
+      code: `import { getCustomTheme } from '@patternfly/react-charts'; const newTheme = getCustomTheme(ChartThemeColor.default, variant, customTheme);`,
+      output: `import { getCustomTheme } from '@patternfly/react-charts'; const newTheme = getCustomTheme(ChartThemeColor.default,  customTheme);`,
```
Comment (gitdallas) [inline packages/eslint-plugin-pf-codemods/test/rules/v5/charts-remove-themeVariant.js:24]:
```suggestion
      output: `import { getCustomTheme } from '@patternfly/react-charts'; const newTheme = getCustomTheme(ChartThemeColor.default, customTheme);`,
```

File: packages/eslint-plugin-pf-codemods/test/rules/v5/charts-remove-themeVariant.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      errors: [
+        {
+          message: `The themeVariant argument has been removed from the react-charts' getCustomTheme function.`,
+          type: "CallExpression",
+        },
+      ],
+    },
+    {
+      code: `import { Chart } from '@patternfly/react-charts'; <Chart themeVariant />`,
+      output: `import { Chart } from '@patternfly/react-charts'; <Chart  />`,
```
Comment (gitdallas) [inline packages/eslint-plugin-pf-codemods/test/rules/v5/charts-remove-themeVariant.js:34]:
```suggestion
      output: `import { Chart } from '@patternfly/react-charts'; <Chart />`,
```

File: packages/eslint-plugin-pf-codemods/lib/rules/v5/charts-remove-darkThemeObjects.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+          "DarkCyanColorTheme",
+          "DarkGoldColorTheme",
+          "DarkGrayColorTheme",
+          "DarkGreenColorTheme",
+          "DarkMultiColorOrderedTheme",
+          "DarkMultiColorUnorderedTheme",
+          "DarkOrangeColorTheme",
+          "DarkPurpleColorTheme",
+        ];
+        const darkThemeImports = node.specifiers.find(
```
Comment (gitdallas) [inline packages/eslint-plugin-pf-codemods/lib/rules/v5/charts-remove-darkThemeObjects.js:18]: technically should be singular

**Review by gitdallas — CHANGES_REQUESTED** [review]
looks good, just some clean up stuff

**Review by gitdallas — CHANGES_REQUESTED** [review]
lgtm, just need to update the readme (extra spaces and commas)

**Review by gitdallas — APPROVED** [review]

**Review by jenny-s51 — APPROVED** [review]
LGTM! Code looks great and mods run as expected.

**Pull request merged**